cup_of_join.py

```diff
@@ -0,0 +1,21 @@
+def join(*lists: list, sep='-'):
+    """
+
+    :param lists: list of lists
+    :param sep: seperator
+    :return: new list with combination of lists with seperator between each
+    two list
+    """
+    if not lists:
+        return None
+
+    ret = []
+    for li in lists:
```

Comment:
read about zip and nested list comprehentions, then refactor this code

Reply:
https://stackoverflow.com/questions/18551458/how-to-frame-two-for-loops-in-list-comprehension-python